### PR TITLE
Changing the execution stage of the gas_validate_function.

### PR DESCRIPTION
**Pull request by stevenlaw123**
Fix for the PR https://github.com/rooch-network/rooch/pull/886.

### CODE REVIEW

File: moveos/moveos/src/moveos.rs [hunk 1]
```diff
@@ -243,12 +268,75 @@ impl MoveOS {
         }
     }
 
-    fn execute_gas_validate(
-        &self,
-        session: &mut MoveOSSession<'_, '_, MoveOSResolverProxy<MoveOSStore>, MoveOSGasMeter>,
-        action: &VerifiedMoveAction,
-    ) -> VMResult<Option<bool>> {
-        match action {
+    pub fn get_address_balance(&self, tx: &MoveOSTransaction) -> VMResult<u128> {
```
Comment (jolestar) [inline moveos/moveos/src/moveos.rs:271]: MoveOs do not know the balance and do not know the `0x3::gas_coin`, so we should migrate this logic to the executor.

File: moveos/moveos/src/moveos.rs [hunk 2]
```diff
@@ -243,12 +268,75 @@ impl MoveOS {
...
+                .expect("the return value of gas validate function should be u128");
+
+                Ok(balance.unchecked_as_u128())
+            }
+            _ => Ok(0),
+        }
+    }
+
+    pub fn execute_gas_validate(&self, tx: &MoveOSTransaction) -> VMResult<Option<bool>> {
+        let MoveOSTransaction {
```
Comment (jolestar) [inline moveos/moveos/src/moveos.rs:322]: Same as above, we can directly call the validate function like the `transaction_validator::validate` in the executor. https://github.com/rooch-network/rooch/blob/6265cd36d2f19e6c42955d53f115dd2dd5af1c73/crates/rooch-types/src/framework/transaction_validator.rs#L29-L57

File: moveos/moveos/src/moveos.rs (hunk 1 above)
Comment (stevenlaw123) [reply moveos/moveos/src/moveos.rs]: OK, it's fixed.

File: moveos/moveos/src/moveos.rs (hunk 2 above)
Comment (stevenlaw123) [reply moveos/moveos/src/moveos.rs]: OK, it's fixed.

**Review by jolestar — APPROVED** [review]
This implementation is still not quite as refined as it could be, so I'll merge it first and refactor it myself.

**Pull request merged**